attributes-next.md

```diff
@@ -0,0 +1,218 @@
+- Deep merging of normal values with default and override is removed (users who were incorrectly using normal
+  as a third precedence level will need to manually fix)
+- Deep merging of automatic values with default/normal/override is removed (this is likely just namespace
+  collision mistakes and stright-up bugs?)
+- The force_default/force_override levels are now the standard behavior of setting attributes in recipe code, the
+  "force" behavior becomes the standard behavior.
+- Environment files consistently have a higher priority / later merge order than roles.
+
+## New Syntax for Ohai Attributes
+
+A new sugar for ohai attributes will be introduced:
+
+```ruby
+ohai("platform")
+ohai("cloud_v2", "public_ipv4_addrs", 0)
+```
+
+This will literally be sugar for the `node.automatic.read("platform")` API that has already been
+shipped.
+
+Write access to ohai data will continue through the existing `node.automatic` API.  In a future RFC
+that will be replaced.
+
+## New Syntax for Merged Default and Override Attributes
+
+```ruby
+attr("foo", "bar", "baz")
+```
+
+This is a reader-only that only reads values merged from default and override levels, with hash-only merging.
+
+It is not intended to be identical to the existing `node["foo"]["bar"]["baz"]` although once deprecation warnings
+against the node object are fixed (see below) they should converge to the same semantics.
+
+The order of merging of the hashes will be:
+
+- default (attribute files)
+- role_default
+- env_default
+- force_default (recipe files)
+- override (attribute files)
+- role_override
+- env_override
+- force_override (recipe files)
+
+Note (below) that attributes written in recipe files will go into the force_default/force_override hashes and will not
+mix with the default/override hashes written in attribute files.  Also note that env_default and role_default have been
+swapped for consistency (the ordering will consistently be attributes/roles/environments/recipes).
+
+# New Syntax for Accessing Default and Override Levels
+
+The writer uses `default_set` and `override_set` while the reader uses simply `default` and `override`.
+
+```ruby
+default_set("foo", "bar", "baz").to("quux")
+default_set("foo", "bar") do |n|
+  n["baz"] = "quux"
+end
+default("foo", "bar", "baz")
+```
+
+In attribute files default and override will be applied to the default and override VividMashes (lowest priority).
+
+In recipe files default and override will be applied to the `force_default` and `force_override` VividMashes.  Setting
+default-level attributes in recipe code will not be supported by the new API.
+
+## New Syntax for Normal Attributes
+
+Since normal attributes have different semantics they will be changed to use `store` and will not deep merge with anything:
+
+```ruby
+store_set("wordpress", "passwd").to("sekret1")
+store_set("wordpress") do |s|
+  s["passwd"] = "sekret1"
+end
+store("wordpress", "passwd")
```

Comment:
`store` is actually a getter, part of the problem is that sounds like its very active verby.  `assign` has the same problem.  `property` collides with the resource concept.